data-products/src/main/scala/org/sunbird/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -0,0 +1,256 @@
+import org.ekstep.analytics.framework.{DruidQueryModel, FrameworkContext, JobConfig, JobContext, StorageConfig, _}
+import org.ekstep.analytics.model.{OutputConfig, ReportConfig}
+
+import java.util.concurrent.atomic.AtomicInteger
+import scala.collection.immutable.{List, Map}
+import org.ekstep.analytics.framework.exception.DruidConfigException
+import org.ekstep.analytics.framework.fetcher.DruidDataFetcher
+import org.ekstep.analytics.framework.util.DatasetUtil.extensions
+import org.ekstep.analytics.model.DruidQueryProcessingModel.{getDateRange, getReportDF}
+import org.apache.hadoop.conf.Configuration
+import org.sunbird.analytics.exhaust.collection.Metrics
+
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.concurrent.CompletableFuture
+import java.util.function.Supplier
+
+
+case class RequestBody(`type`: String,`params`: Map[String,AnyRef])
+case class OnDemandDruidResponse(file: List[String], status: String, statusMsg: String, execTime: Long)
+
+object OnDemandDruidExhaustJob extends optional.Application with BaseReportsJob with Serializable with OnDemandExhaustJob{
+  override def getClassName() : String = "org.sunbird.analytics.exhaust.OnDemandDruidExhaustJob"
+  val jobName: String = "OnDemandDruidExhaustJob"
+  def name(): String = "OnDemandDruidExhaustJob"
+  // $COVERAGE-OFF$ Disabling scoverage for main and execute method
+  def main(config: String)(implicit sc: Option[SparkContext] = None, fc: Option[FrameworkContext] = None) {
+    JobLogger.init("OnDemandDruidExhaustJob")
+    JobLogger.start("OnDemandDruidExhaustJob Started executing", Option(Map("config" -> config, "model" -> name)))
+    implicit val jobConfig = JSONUtils.deserialize[JobConfig](config)
+    implicit val spark: SparkSession = openSparkSession(jobConfig)
+    implicit val sc: SparkContext = spark.sparkContext
+    implicit val sqlContext = new SQLContext(sc)
+    JobContext.parallelization = CommonUtil.getParallelization(jobConfig)
+
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    implicit val conf = spark.sparkContext.hadoopConfiguration
+    execute()
+    // $COVERAGE-ON$ Disabling scoverage for main and execute method
+  }
+
+  def validateEncryptionRequest(request: JobRequest): Boolean = {
+    if (request.encryption_key.nonEmpty) true else false;
+  }
+
+  def validateRequest(request: JobRequest): Boolean = {
+    if (request.request_data != null) true else false
+  }
+
+  def markRequestAsProcessing(request: JobRequest):Boolean = {
+    request.status = "PROCESSING";
+    updateStatus(request);
+  }
+
+  def getStringProperty(config: Map[String, AnyRef], key: String, defaultValue: String) : String = {
+    config.getOrElse(key, defaultValue).asInstanceOf[String]
+  }
+  def druidAlgorithm(reportConfig: ReportConfig)(implicit spark: SparkSession, sqlContext: SQLContext, fc: FrameworkContext, sc:SparkContext, config: JobConfig): RDD[DruidOutput]  ={
```

Comment:
One implicit sparkSession is sufficient. sqlContext and sparkContext are redundant

Reply:
sqlContext is removed , But implicit sparkContext is required to call DruidDataFetcher.getDruidData()